toolkit/scripts/toolchain/build_official_toolchain_rpms.sh

```diff
@@ -390,6 +390,20 @@ chroot_and_install_rpms zlib
 build_rpm_in_chroot_no_install perl
 chroot_and_install_rpms perl
 
+# perl-generators requires perl-Fedora-VSP
+# All perl packages need perl-generators to correctly
+# generate their run-time provides and requires.
+build_rpm_in_chroot_no_install perl-Fedora-VSP
+chroot_and_install_rpms perl-Fedora-VSP
+build_rpm_in_chroot_no_install perl-generators
+chroot_and_install_rpms perl-generators
+
+# Rebuilding perl packages with perl-generators installed.
```

Comment:
I feel like we might have an issue here with the delta build... lets assume `pkg.rpm` is NOT available for delta, but we are running a delta build otherwise.

We place our prospective delta build candidates into `$CHROOT_RPMS_DIR` on line 125.

In `build_rpm_in_chroot_no_install()` we check if we have a delta rpm via `$(find $CHROOT_RPMS_DIR -name "$rpm" -print -quit)`. If we are missing one (as is our case here), we will run `rpmbuild` which I believe also places files into ` $CHROOT_RPMS_DIR`.

We then collect our final product via `copy_built_rpms()` which takes them from `$CHROOT_RPMS_DIR` to `$FINISHED_RPM_DIR`.

When we now run the second build, I'd expect that `$(find $CHROOT_RPMS_DIR -name "$rpm" -print -quit)` will now see the first run files already present, and skip the second build.

Reply:
https://github.com/microsoft/azurelinux/pull/9264 should fix

Reply:
Merged, thank you, @dmcilvaney!